src/nbl/asset/utils/CIESProfile.h

```diff
@@ -0,0 +1,137 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef __NBL_ASSET_C_IES_PROFILE_H_INCLUDED__
+#define __NBL_ASSET_C_IES_PROFILE_H_INCLUDED__
+
+#include "nbl/asset/metadata/CIESProfileMetadata.h"
+#include "nbl/core/Types.h"
+#include <sstream>
+
+namespace nbl 
+{
+    namespace asset 
+    {
+        class CIESProfile 
+        {
+            public:
+                using IES_STORAGE_FORMAT = double;
+
+                _NBL_STATIC_INLINE_CONSTEXPR size_t CDC_DEFAULT_TEXTURE_WIDTH = 1024;
+                _NBL_STATIC_INLINE_CONSTEXPR size_t CDC_DEFAULT_TEXTURE_HEIGHT = 1024;
+
+                _NBL_STATIC_INLINE_CONSTEXPR IES_STORAGE_FORMAT MAX_VANGLE = 180.0;
+                _NBL_STATIC_INLINE_CONSTEXPR IES_STORAGE_FORMAT MAX_HANGLE = 360.0;
+
+                _NBL_STATIC_INLINE_CONSTEXPR auto UI16_MAX_D = 65535.0;
+                _NBL_STATIC_INLINE_CONSTEXPR auto IES_TEXTURE_STORAGE_FORMAT = asset::EF_R16_UNORM;
+
+                enum Version : uint8_t
+                {
+                    V_1995,
+                    V_2002,
+                    V_SIZE
+                };
+
+                enum PhotometricType : uint8_t 
+                {
+                    TYPE_NONE,
+                    TYPE_C,
+                    TYPE_B,
+                    TYPE_A
+                };
+
+                enum LuminairePlanesSymmetry : uint8_t
+                {
+                    ISOTROPIC,                  //! Only one horizontal angle present and a luminaire is assumed to be laterally axial symmetric
+                    QUAD_SYMETRIC,              //! The luminaire is assumed to be symmetric in each quadrant
+                    HALF_SYMETRIC,              //! The luminaire is assumed to be symmetric about the 0 to 180 degree plane
+                    OTHER_HALF_SYMMETRIC,       //! HALF_SYMETRIC case for legacy V_1995 version where horizontal angles are in range [90, 270], in that case the parser patches horizontal angles to be HALF_SYMETRIC
+                    NO_LATERAL_SYMMET           //! The luminaire is assumed to exhibit no lateral symmet
+                };
+
+                CIESProfile() = default;
+                ~CIESProfile() = default;
+
+                const IES_STORAGE_FORMAT& getHAnglesOffset() const { return hAnglesOffset; }
```

Comment:
why do you need to store `hAnglesOffset` wouldn't it be either 0 or 90 depending on `OTHER_HALF_SYMMETRIC` ?